bal_tools/subgraph.py

```diff
@@ -95,6 +97,7 @@ def get_subgraph_url_frontendv2(self, subgraph):
                         if urlparse(url).scheme in ["http", "https"]:
                             graph_api_key = os.getenv("GRAPH_API_KEY")
                             if "${keys.graph}" in url and not graph_api_key:
+                                warnings.warn(f"`GRAPH_API_KEY` not set. may be rate limited for {url}")
```

Comment:
The problem we had was that the dev nodes may also be quite out of date.  Maybe change this 
```suggestion
                                warnings.warn(f"`GRAPH_API_KEY` not set. may be rate limited or have stale data for subgraph:{subgraph} url:{url}")
```